src/eba_view.eliom

```diff
@@ -190,4 +190,191 @@ let password_form ?a ~service () =
          string_input ~input_type:`Submit ~value:"Send" ()
        ])
     ()
- }}
+
+let multiple_email_div_id = "multiple_email"
+}}
+{server{
+let rpc_emails_and_params_of_userid =
+  server_function Json.t<int64> Eba_user.emails_and_params_of_userid
+
+let rpc_add_email_to_user =
+  let add_email (user, email) =
+      Eba_user.add_email_to_user user email
+  in
+  server_function Json.t<int64 * string> add_email
+
+let rpc_update_users_primary_email =
+  server_function Json.t<string> Eba_user.update_users_primary_email
+
+let rpc_delete_email =
+  server_function Json.t<string> Eba_user.delete_email
+
+let rpc_send_mail_confirmation =
+  server_function Json.t<string> Eba_user.send_mail_confirmation
+}}
+
+{client{
+
+module ReactList = struct
+    let list t =
+      let open ReactiveData.RList in
+      make_from
+        (React.S.value t)
+        (React.E.map (fun e -> Set e) (React.S.changes t))
+end
+
+module Model = struct
+    type activation_state = [`Unit | `Act_key_sent | `Activated]
+    type non_primary_mail = {
+        email: string;
+        activation_state: activation_state;
+      }
+    type mails = {userid: int64;
+                  add_mail_error: string option;
+                  primary: string;
+                  others: non_primary_mail list}
+    type state = [`NotConnected | `Connected of mails]
+    type rs = state React.signal
+    type rf = ?step:React.step -> state -> unit
+    type rp = rs * rf
+
+    let create_connected userid =
+      lwt mails = %rpc_emails_and_params_of_userid userid in
+      let f (_, primary, _) = primary in
+      let primary, others =  List.partition f mails in
+      let (primary, _, _) = List.hd primary in
+      let add_mail_error = None in
+      let others = List.map
+                     (fun (email, _, is_activated) ->
+                      let activation_state =
+                        if is_activated then `Activated
+                        else `Unit
+                      in
+                      {email;
+                       activation_state})
+                     others in
+      Lwt.return {userid;add_mail_error;
+                  primary; others}
+
+    let create userido =
+      match userido with
+      | None -> Lwt.return `NotConnected
+      | Some userid ->
+         lwt res = (create_connected userid) in
+         Lwt.return (`Connected res)
+end
+
+let create_input name =
+  Tyxml_js.Html5.(input ~a:[a_input_type `Text; a_placeholder name]) ()
```

Comment:
See ["generating HTML pages"](https://ocsigen.org/eliom/dev/manual/clientserver-html) in the manual, and especially the ["reactive DOM"](https://ocsigen.org/eliom/dev/manual/clientserver-html#reactive) part.

Similar functionality is now available for code inside server sections. There is a [tutorial for that](https://ocsigen.org/tuto/4.2/manual/tutoreact) which describes the basic ideas, but the module names are outdated (`Eliom_csreact` -> `Eliom_shared`, `Eliom_csreact.R` -> `Eliom_content.Html5.R`, plus internal renaming in `Eliom_shared`). It would be nice to use these server-side modules for the new stuff that goes in Eliom-Base-App.